python/lsst/verify/compatibility/metricTask.py

```diff
@@ -0,0 +1,196 @@
+            are DataIds (or lists of DataIds) that task consumes for
+            corresponding dataset type. DataIds are guaranteed to match data
+            objects in ``inputData``
+        outputDataId : `dict` from `str` to dataId
+            Dictionary containing a single key-value pair, from
+            ``"measurement"`` to a single data ID for the measurement. The
+            data ID must have the same granularity as the metric.
+
+        Returns
+        -------
+        struct : `lsst.pipe.base.Struct`
+            A ``Struct`` containing at least the following component:
+
+            - ``measurement``: the value of `getOutputMetric` derived from
+                               ``inputData`` (`lsst.verify.Measurement` or
+                               `None`). The measurement shall contain not only
+                               the value of the metric, but any additional
+                               data needed to place it in the proper context.
+
+        Raises
+        ------
+        lsst.verify.MetricComputationError
+            Raised if an algorithmic or system error prevents calculation
+            of the metric. Examples include corrupted input data or
+            unavoidable exceptions raised by analysis code. The
+            ``MetricComputationError`` should be chained to a more specific
+            exception describing the root cause.
+
+            Note that not having enough data for a metric to be applicable is
+            not an error, and should not trigger this exception.
+
+        Examples
+        --------
+        Consider a metric that characterizes PSF variations across the entire
+        field of view, given processed images. Then, if ``run`` has the
+        signature ``run(images)``:
+
+        .. code-block:: py
+
+            inputData = {'images': [image1, image2, ...]}
+            inputDataIds = {'images': [{'visit': 42, 'ccd': 1},
+                                       {'visit': 42, 'ccd': 2},
+                                       ...]}
+            outputDataId = {'measurement': {'visit': 42}}
+        """
+        result = self.run(**inputData)
+        if result.measurement is not None:
+            self.addStandardMetadata(result.measurement,
+                                     outputDataId["measurement"])
+        return result
+
+    @classmethod
+    @abc.abstractmethod
+    def getInputDatasetTypes(cls, config):
+        """Return input dataset types for this task.
+
+        Parameters
+        ----------
+        config : ``cls.ConfigClass``
+            Configuration for this task.
+
+        Returns
+        -------
+        datasets : `dict` from `str` to `str`
+            Dictionary where the key is the name of the input dataset (must
+            match a parameter to `run`) and the value is the name of its
+            Butler dataset type.
+        """
+
+    @classmethod
+    @abc.abstractmethod
+    def getOutputMetric(cls, config):
```

Comment:
`getOutputMetric` makes it sound like this returns a `lsst.verify.Metric` instance, but it's only returning a name. I'd recommend renaming the method to `getOutputMetricName`. And since "output" doesn't add much, perhaps just: `getMetricName`?

Reply:
The name was a deliberate imitation of [`PipelineTask.getOutputDatasetTypes`](https://pipelines.lsst.io/v/daily/py-api/lsst.pipe.base.PipelineTask.html#lsst.pipe.base.PipelineTask.getOutputDatasetTypes) (which may need to be implemented in terms of this method, depending on how Butler persistence of measurements ends up working). I could do `getOutputMetricName`, but dropping the `Output` may be confusing later.

Reply:
I see. I'd recommend putting name in the title, but that's just me. :)